lib/spack/spack/solver/asp.py

```diff
@@ -2072,7 +2072,7 @@ def _supported_targets(self, compiler_name, compiler_version, targets):
             try:
                 with warnings.catch_warnings():
                     warnings.simplefilter("ignore")
-                    target.optimization_flags(compiler_name, compiler_version)
+                    target.optimization_flags(compiler_name, str(compiler_version))
```

Comment:
?

Reply:
This is because of https://github.com/archspec/archspec/pull/155 If we had mypy checks active we would have caught this before, but `archspec` accepts strings as argument - and it doesn't know about Spack `StandardVersion`